packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -1545,10 +1546,18 @@ class NavigatorState extends State<Navigator> with TickerProviderStateMixin {
     for (NavigatorObserver observer in widget.observers)
       observer.didPush(route, oldRoute);
     assert(() { _debugLocked = false; return true; }());
-    _cancelActivePointers();
+    _afterNavigation();
     return route.popped;
   }
 
+  void _afterNavigation() {
+    const bool isReleaseMode = bool.fromEnvironment('dart.vm.product');
+    if (!isReleaseMode) {
+      developer.postEvent('Flutter.Navigation', <String, dynamic>{});
```

Comment:
Maybe add a quick comment for context what these events are used for?

Reply:
added a comment.
      // This event is used by performance tools that show stats for the
      // time interval since the last navigation event occurred ensuring that
      // stats only reflect the current page.
      // These tools do not need to know exactly what the new route is so no
      // attempt is made to describe the current route as part of the event.